src/tests/nativeaot/SmokeTests/HardwareIntrinsics/x64Vex.csproj

```
@@ -31,6 +31,15 @@ $(BashCLRTestPreCommands)
           exit 0
         fi
     fi
+    if [[ "$OSTYPE" == "linux"* ]]; then
+        more /proc/cpuinfo
+
+        grep avx2 /proc/cpuinfo >/dev/null
+        if [ $? -ne 0 ]; then
```

nit: we normally use an inline command in situations where we only care about its success/failure. In case of grep command, we use `grep -q` which transforms pattern matching into "if contains" semantics. Also, a more targeted pattern would be nice:
```suggestion
        if ! grep -q '^flags.*avx2' /proc/cpuinfo 2>/dev/null; then
```
(this also avoids other errors on stderr, e.g. when procfs is not available or accessible)

---

I really dislike this kind of woodo in Bash only because I cannot internalize proper syntax for `if` and differences `[` and `[[`. But that's on me. Hopefully that's make PR pass.

---

Yea, I know it's legit to bash bash(1). :)

> For commands success/failure, prefer `if command; then` over `[[ "$?"`. Always prefer `[[` over `[` in bash context. Avoid awk (that has like seven different flavors..) usages in parsing/capturing-groups situations and instead use bash pattern matching. Quote variables to avoid expansion issues (shellcheck linter helps figuring out all such cases). Choose bash features where possible; before delegation to external commands (simple grep is ok, anything requiring perl extensions etc. can hamper portability effort). Stay compatible with bash v3.2 (i.e. last version before bash changed their license and Apple didn't upgraded bash).

These are a few things which we have tried to pay specific attention to in previous refactorings of bash scripts; as bash is the primarily scripting language used in runtime and arcade for Unix -- and there are significant LOC.